network/src/protocols/rpc/mod.rs

```diff
@@ -206,71 +200,33 @@ where
     /// Start the [`Rpc`] actor's event loop.
     pub async fn start(self) {
```

Comment:
what's the advantage of having async workers / tasks like this per peer, it would seem we could make this synchronous unless you are leveraging the task to maintain state (as we did in the past)

Reply:
right! I was actually proposing to @bothra90 yesterday that the DirectSend and Rpc actors don't need to be seperate and could be owned by the `Peer` actor instead who just static dispatches substreams to each